pkg/sources/source_manager.go

```diff
@@ -56,12 +65,19 @@ func WithBufferedOutput(size int) func(*SourceManager) {
 	return func(mgr *SourceManager) { mgr.outputChunks = make(chan *Chunk, size) }
 }
 
+// WithSourceUnits enables using source unit enumeration and chunking if the
+// source supports it.
+func WithSourceUnits() func(*SourceManager) {
+	return func(mgr *SourceManager) { mgr.useSourceUnits = true }
+}
+
 // NewManager creates a new manager with the provided options.
 func NewManager(opts ...func(*SourceManager)) *SourceManager {
 	mgr := SourceManager{
 		// Default to the headless API. Can be overwritten by the WithAPI option.
 		api:          &headlessAPI{},
 		handles:      make(map[handle]SourceInitFunc),
+		reports:      make(map[handle][]*JobReport),
```

Comment:
yeah that's a good point.. I think I'll leave it separate for now since the use cases are different. We aren't always accessing both at the same time. If we add a third `map[handle]T` I think that's a good point to aggregate them.